routes/route_team.jsx

```diff
@@ -83,6 +83,10 @@ const WAKEUP_CHECK_INTERVAL = 30000; // 30 seconds
 const WAKEUP_THRESHOLD = 60000; // 60 seconds
 
 function preNeedsTeam(nextState, replace, callback) {
+    if (nextState.location.action === 'REPLACE') {
```

Comment:
Is this only going to be `true` if we were previously on a team? What happens if we're on a non-team page such as the system console and then switch to a team? Just making sure we're not going to miss out on any required requests in that case

Reply:
From a non-team or even team page, this condition would be `false`(action will be `PUSH`) the first time you enter `team url` route- triggering requests and actions. If there is no channel param then we `replace` with a channel param and triggering a `REPLACE` action.
Only in the above scenario, this skips these actions and network calls.

Reply:
I went through different routes and checked PR seems to be working fine. Let me know if I am missing anything.

Reply:
Ok cool 👍